CIScripts/Common/Invoke-UntilSucceeds.ps1

```diff
@@ -0,0 +1,56 @@
+. $PSScriptRoot\Aliases.ps1
+. $PSScriptRoot\Exceptions.ps1
+
+function Invoke-UntilSucceeds {
+    <#
+    .SYNOPSIS
+    Utility wrapper for making sure that the assert is consistently true. It works by retrying
+    the assert every Interval seconds, up to Duration.
+    .PARAMETER ScriptBlock
+    ScriptBlock containing a Pester assertion.
+    .PARAMETER Interval
+    Interval (in seconds) between retries.
+    .PARAMETER Duration
+    Timeout (in seconds).
+    #>
+    Param (
+        [Parameter(Mandatory=$true,
+                   ValueFromPipeline=$true)] [ScriptBlock] $ScriptBlock,
+        [Parameter(Mandatory=$false)] [int] $Interval = 1,
+        [Parameter(Mandatory=$false)] [int] $Duration = 3
+    )
+    if ($Duration -lt $Interval) {
+        throw "Duration must be longer than interval"
+    }
+    if ($Interval -eq 0) {
+        throw "Interval must not be equal to zero"
+    }
+    $HasPrintedDots = $false
+    $StartTime = Get-Date
+    while (((Get-Date) - $StartTime).Seconds -lt $Duration) {
+        try {
+            $ReturnVal = & $ScriptBlock
+            if ($ReturnVal) {
+                break
+            } else {
+                throw New-Object -TypeName CITimeoutException("Did not evaluate to True." + 
+                    "Last return value encountered was: $ReturnVal.")
+            }
+        } catch {
+            $HasPrintedDots = $true
+            Write-Host -NoNewline "."
+            $LastException = $_.Exception
+            Start-Sleep -s $Interval
+        }
+    }
+    if ($HasPrintedDots) {
+        Write-Host ""
+    }
```

Comment:
We think that printing dots is not a responsibility of this function. If somebody wants to print them, they can add `Write-Host` to script block.